### PR TITLE
Robot moves on to do list

### PR DESCRIPTION
**Pull request by barabashlyuda77**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -9,5 +9,11 @@
       "sinon": true,
       "proxyquire": true,
       "expect": true
-  }
+  },
+  "rules": {
+        "no-param-reassign": ["error", { "props": false }],
+        "no-plusplus": ["error", { "allowForLoopAfterthoughts": true }],
+        "no-console": "off",
+        "max-len": "off"
```
Comment (GContaldi) [inline .eslintrc.json]: it's important to break down

File: src/robot.js [hunk 2]
```diff
@@ -1,5 +1,22 @@
 const _ = require('lodash');
+const { areTasksToWorkOn, updateTasksStage, randomLocation } = require('./task');
 
-const move = (map, currentLocation) => _.sample(map[currentLocation]);
+const move = (map, location) => _.sample(map[location]);
 
-module.exports = { move };
+const works = (tasks, map) => {
+  let turn;
+  let currentLocation = randomLocation(map);
```
Comment (GContaldi) [inline src/robot.js]: maybe this is something for map

File: src/task.js [hunk 3]
```diff
@@ -1,21 +1,65 @@
...
+  task.stage = 'started';
+};
+
+const finishTask = (task) => {
+  task.stage = 'finished';
+};
+
+const isTaskStarted = task => task.stage === 'started';
+
+const isUnstartedTask = task => task.stage === 'unstarted';
```
Comment (GContaldi) [inline src/task.js]: maybe we can use the same pattern to call the method, I mean `isTaskUnstarted`

File: src/task.js [hunk 4]
```diff
@@ -1,21 +1,65 @@
...
+
+const startTask = (task) => {
+  task.stage = 'started';
+};
+
+const finishTask = (task) => {
+  task.stage = 'finished';
+};
+
+const isTaskStarted = task => task.stage === 'started';
```
Comment (GContaldi) [inline src/task.js]: 'started' and all other states, can become a constants. e.g.
```js
const STARTED = 'started';
...
// or
const states = { STARTED: 'started', ... }
```

File: src/task.js [hunk 5]
```diff
@@ -1,21 +1,65 @@
...
+    }
+    if (isTaskStartedAtLocation(task, location)) {
+      finishTask(task);
+      console.log(`Finished task at ${task.to}`);
+    }
   });
-}
+};
+
+const areTasksToWorkOn = tasks => tasks.some(task => task.stage !== 'finished');
```
Comment (GContaldi) [inline src/task.js]: maybe
```js
const isTaskFinished = task => task.stage === 'finished';
const areTasksToWorkOn = tasks => tasks.some(task => !isTaskFinished(task));
```

**Comment by barabashlyuda77** [comment]
@GContaldi  please check changed I've made

File: .eslintrc.json [hunk 6]
```diff
@@ -9,5 +9,9 @@
       "sinon": true,
       "proxyquire": true,
       "expect": true
-  }
+  },
+  "rules": {
+        "no-plusplus": ["error", { "allowForLoopAfterthoughts": true }],
```
Comment (GContaldi) [inline .eslintrc.json]: too much tabbing :)

**Review by GContaldi — APPROVED** [review]

**Comment by GContaldi** [comment]
Nicely done!

**Pull request merged**